pages/en/lb4/Decorator.md

```diff
@@ -0,0 +1,196 @@
+
+As a default, LoopBack comes with some pre-defined component decorators for your use:
+- [authentication](#authentication-decorators)
+- [repository](#repository-decorators)
+
+## Route Decorators
+- @get, @post, @put, @patch, @del
+
+  Register route for a controller by rest decorator, for their usage, refer to [Routing to Controllers](controllers.htm#Routing-to-Controllers)
+
+- @api
+
+  Register route for a controller by api decorator, more details and example refer to [Specifying Controller APIs](controllers.htm#Specifying-Controller-APIs)
+
+- @operation
+
+  Expose a Controller method as a REST API operation. Usage refer to [TBD in Controllers](controllers.htm#)
+
+- @param
+
+  Describe an input parameter of a Controller method. 
+  
+  `@param` can be applied to method itself or specific parameters. For example,
+
+  ```ts
+    class MyController {
+      @get('/')
+      @param(offsetSpec)
+      @param(pageSizeSpec)
+      list(offset?: number, pageSize?: number) {}
+    }
+  ```
+
+  or
+  
+  ```ts
+    class MyController {
+      @get('/')
+      list(
+        @param(offsetSpec) offset?: number,
+        @param(pageSizeSpec) pageSize?: number,
+      ) {}
+    }
+  ```
+  
+  More example can be found in [Writing Controller methods](controller.htm#Writing-Controller-methods)
+  
+## Dependency Injection
+
+@inject is a decorator to annotate method arguments for automatic injection by LoopBack IoC container. It can only be applied on a **method parameter** or a **property**.
+
+The `@inject` decorator allows you inject dependencies bound to the application context, or
+to any implementation of the [Context](#context) object. You can bind values, class definitions and 
+provider functions to those contexts and then resolve values (or the results of functions that return those 
+values!) in other areas of your code.
+
+```ts
+// application.ts
+import {Application} from '@loopback/core';
+import 'fs-extra';
+class MyApp extends Application {
+  constructor() {
+    super();
+    const app = this;
+    const widgetConf = JSON.parse(fs.readSync('./widget-config.json'));
+    app.bind('config.widget').to(widgetConf);
+  }
+}
+```
+Now that we've bound the 'config.widget' key to our configuration object, we can inject it in our WidgetController:
+
+```ts
+// widget-controller.ts
+import {widgetSpec} from '../apispec';
+@api(widgetSpec)
+class WidgetController {
+  @inject('config.widget')
+  protected config: JSON; // This will be resolved at runtime!
+
+  // etc...
+}
+```
+
+It's been documented as a LoopBack core concept in [Dependency Injection](Dependency-Injection.htm)
+For more information, see the [Dependency Injection](Dependency-Injection.htm) section under [LoopBack Core Concepts](Concepts.htm)
+
+
+## Authentication Decorator
+
+- @authenticate
+
+  Mark a controller method as requiring authenticated user, it takes in a strategy name. An example is using BasicStrategy to authenticate user in function `whoAmI`:
```

Comment:
"It takes in a strategy name" should be reworded to be explicit; mention that the decorator takes a strategy name as a parameter, and make sure there is an explanation for what that is (or a link to somewhere else that does explain it)

Reply:
yep I think the snippet below explains it